action.yml

```diff
@@ -8,11 +8,49 @@ inputs:
   description:
     type: string
     required: true
+  close_existing:
+    type: boolean
+    required: false
+    default: false
 
 runs:
   using: composite
   steps:
 
+    - name: Close existing issues
+      shell: bash
+      if: ${{ inputs.close_existing == 'true' }}
+      env:
+        TITLE: ${{ inputs.title }}
+        GITHUB_REPOSITORY: ${{ github.repository }}
+      run: |
+        # Close any existing issues with the same title
+        # https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28
+        RESP_CODE=$(curl -w %{http_code} -s -o __issues.json \
+        -X GET https://api.github.com/repos/$GITHUB_REPOSITORY/issues?state=open \
+        -H "Accept: application/vnd.github+json" \
+        -H "Authorization: token ${{ github.token }}" \
+        )
+        if [[ $RESP_CODE != "200" ]]; then
+          echo "Unable to fetch issues - HTTP response code was $RESP_CODE"
+          exit 1
+        fi
+        ISSUE_NUMBERS=$(jq -r --arg TITLE "$TITLE" '.[] | select(.title == $TITLE) | .number' __issues.json)
```

Comment:
If feasible, we should check the user that created the issue as well. We only want to close auto-created issues.

Reply:
Done